power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -284,9 +285,6 @@ inline void process_edges_dijkstra(Idx v, std::vector<EdgeWeight>& vertex_distan
             if (u == s && vertex_distances[s] + weight < vertex_distances[t]) {
                 vertex_distances[t] = vertex_distances[s] + weight;
                 pq.emplace(vertex_distances[t], t);
-            } else if (u == t && vertex_distances[t] + weight < vertex_distances[s]) {
-                vertex_distances[s] = vertex_distances[t] + weight;
-                pq.emplace(vertex_distances[s], s);
```

Comment:
please also check if you can remove the above comment
```
            // We can not use BGL_FORALL_OUTEDGES here because we need information
            // regardless of edge direction
```
and/or even move to `BGL_FORALL_OUTEDGES` altogether